### DISCUSSION TITLE
Wildcards in automation / Amazon description spam

### DISCUSSION THREAD

**Discussion by alexchatwin** [firefly-iii/firefly-iii#8625] [Firefly III] [answered]
![image](https://github.com/firefly-iii/firefly-iii/assets/22892404/2e32e626-e110-4693-9079-73d82bec1d41)
I have 2 problems (I import from gocardless)
1) I get a load of junk data from Amazon in my import, I can search for these in the UI using 'am z' but this isn't accepted in the automations. I've got a work-around where I manually tag the transactions, then automate moving them all to a single Amazon account.. but then
2) I'm left with loads of useless expense accounts like:
![image](https://github.com/firefly-iii/firefly-iii/assets/22892404/af2c4486-7ebb-4ffc-8086-eb35ad4b1f58)

Any ideas welcome!

**Comment by JC5** [comment, marked as answer]
Create two rules, something like:

- Rule 1 triggers on "description contains amazon" and "description is NOT exactly amazon".
- Rule 2 triggers on "description contains amzn"

Both have the action "set destination account Amazon".

That should fix it. You still end up with a bunch of unused expense accounts but you can flush those from your profile.

**Reply by alexchatwin** [reply]
Thanks, and I learned about purging unused accounts
